service/tablet_allocator.cc

```diff
@@ -290,8 +399,9 @@ class load_balancer {
     }
 
 public:
-    load_balancer(token_metadata_ptr tm, locator::load_stats_ptr table_load_stats, load_balancer_stats_manager& stats)
-        : _tm(std::move(tm))
+    load_balancer(token_metadata_ptr tm, locator::load_stats_ptr table_load_stats, load_balancer_stats_manager& stats, uint64_t target_max_tablet_size)
```

Comment:
I think it's nicer to target the average tablet size, not the maximum. Then the boundaries are a factor of two away from either direction.

Reply:
(and it's easier to reason about the average number of tablet replicas per shard)

Reply:
it's essentially what we do now, but you want that explicit in the code by having a target_tablet_size config, rather than max_tablet_size, which we infer target_tablet_size from, right?

Reply:
maybe, for the end user, it's indeed easier to reason about target_tablet_size, but the good thing about max_tablet_size is that it's more explicit how much an individual tablet can grow to.